modules/actions/workflows.go

```diff
@@ -240,7 +240,7 @@ func matchPushEvent(commit *git.Commit, pushPayload *api.PushPayload, evt *jobpa
 			if err != nil {
 				break
 			}
-			if !workflowpattern.Filter(patterns, []string{refName.ShortName()}, &workflowpattern.EmptyTraceWriter{}) {
+			if !workflowpattern.Filter(patterns, []string{refName.TagName()}, &workflowpattern.EmptyTraceWriter{}) {
```

Comment:
I guess here (and above) it needs to check `IsBranch` or `IsTag` first?

Otherwise, if the pattern is `*`, then a branch's TagName returns `""`, it matches the pattern `*` ?

(Maybe a legacy problem)

Reply:
This has been checked, see line 236